src/lib.rs

```diff
@@ -479,10 +521,12 @@ pub fn run(
             .unwrap_or_else(|| Target::from(host.triple(), &target_list));
         config.confusable_target(&target, msg_info)?;
 
+        let uses_zig = config.zig(&target).unwrap_or(false);
+        let zig_version = config.zig_version(&target)?;
         // Get the image we're supposed to base all our next actions on.
         // The image we actually run in might get changed with
         // `target.{{TARGET}}.dockerfile` or `target.{{TARGET}}.pre-build`
-        let image = match docker::get_image(&config, &target) {
+        let image = match docker::get_image(&config, &target, uses_zig) {
```

Comment:
I mean, the image needs to be dependent on zig, because if it's present, we use a different image. Also, we can't have a manual image for a given target and zig support, because we can't know outside the function without just redoing the logic if the returned image was from the config/environment or just the default from the target.

Reply:
I think keeping pre-build hooks is fine: it doesn't have any real issues with `zig cc` itself: it has a toolchain itself installed, can dynamically to libraries installed by a package manager, etc.